lib/internal/Magento/Framework/Session/SessionManager.php

```diff
@@ -501,7 +514,32 @@ public function regenerateId()
             return $this;
         }
 
-        $this->isSessionExists() ? session_regenerate_id(true) : session_start();
+        if ($this->isSessionExists()) {
+            // Regenerate the session
+            session_regenerate_id();
+            $newSessionId = session_id();
+
+            $_SESSION['new_session_id'] = $newSessionId;
+
+            // Set destroy timestamp
+            $_SESSION['destroyed'] = time();
+
+            // Write and close current session;
+            session_commit();
+            // Called after destroy()
+            $oldSession = $_SESSION;
+            // Start session with new session ID
+            session_id($newSessionId);
+            ini_set('session.use_strict_mode', 0);
```

Comment:
Do you have any examples when manipulating strict mode on the fly is required? I've tested solution briefly with session strict mode enabled and these lines commented and have not seen any issue with this so far.

Reply:
It's because is how is shown in the example 2 here http://php.net/manual/en/function.session-regenerate-id.php. Actually now that I think about the line 532 and 533 should be swapped. However you are right, seems not useful: even with strict_mode enabled I'm able to set the session_id from $newSessionId. Maybe @acetronaut has further thoughts about.